### PR TITLE
feat(gitlab): add merge method check

### LINKED ISSUE #5565: Autodetect GitLab fast-forward branch controls
**What would you like Renovate to be able to do?**

Detect this setting:
image

And if `rebaseWhen=auto` then change this to `rebaseWhen=behind-base-branch`, like we do with GitHub's branch protection.

**Describe the solution you'd like**

Exactly like the same in GitHub

**Describe alternatives you've considered**

Right now attempted automerges of Renovate MRs that are behind base branch fail with a 406 error.

**Additional context**

https://github.com/renovatebot/config-help/issues/549

### PR DESCRIPTION
**Pull request by viceice**
If `merge_method`is not `merge` we need to force rebase our branches.

Closes #5565

### CODE REVIEW

**Comment by viceice** [comment]
No live test yet. Maybe safe without?

File: lib/platform/gitlab/index.ts [hunk 1]
```diff
@@ -259,8 +264,8 @@ export async function initRepo({
   return repoConfig;
 }
 
-export function getRepoForceRebase(): boolean {
-  return false;
+export function getRepoForceRebase(): Promise<boolean> {
+  return Promise.resolve(config?.mergeMethod !== 'merge');
```
Comment (rarkins) [inline lib/platform/gitlab/index.ts]: rebase merge doesn't work?

Comment (viceice) [reply lib/platform/gitlab/index.ts]: Nope, you have to rebase before merge too.

**Comment by rarkins** [comment]
I just realized something - we need to check when this field was introduced. Otherwise if there's older GitLab versions which don't support the field then they'll start always rebasing.

**Comment by viceice** [comment]
ok, i'll look at the api

**Comment by viceice** [comment]
https://docs.gitlab.com/ee/policy/maintenance.html

looks like they only support the latest 3 versions.

I can't find when try introduced the property, but i think it was long time ago. (at least readonly)

but to be safe, we can check existance.

**Comment by viceice** [comment]
looks like https://gitlab.com/gitlab-org/gitlab-foss/-/merge_requests/14272 introduced it in version 10.1.

Older versions have had other properties. Should we support older versions than 10.1, which are no longer supported?

**Comment by rarkins** [comment]
We should support whatever GitLab says is supported, but ideally not break older stuff also if we can avoid it.

**Comment by viceice** [comment]
We do not break it, we only force a rebase if we fall behind on version less than 10.1 🙈

**Comment by viceice** [comment]
a simple fix for older versions would be `config.mergeMethod = res.body.merge_method || 'merge';`

**Pull request merged**